spyder_notebook/notebookplugin.py

```diff
@@ -223,13 +230,44 @@ def close_client(self, index=None, client=None):
         if index is not None:
             client = self.tabwidget.widget(index)
 
+        self.shutdown_kernel(client)
+
         # TODO: Eliminate the notebook from disk if it's an Untitled one
         client.close()
 
         # Note: notebook index may have changed after closing related widgets
         self.tabwidget.removeTab(self.tabwidget.indexOf(client))
         self.clients.remove(client)
 
+    def shutdown_kernel(self, client):
+        """Shutdown the kernel of the given client."""
+        sessions_url = url_path_join(client.server_url, 'api/sessions')
+        sessions_req = requests.get(sessions_url).content.decode()
+        sessions = json.loads(sessions_req)
+        kernel_id = None
+        if os.name == 'nt':
+            path = client.path.replace('\\', '/')
+        else:
+            path = client.path
+        for session in sessions:
+            if session['notebook']['path'] == path:
+                kernel_id = session['kernel']['id']
+                break
+        if kernel_id:
+            delete_url = url_path_join(client.server_url,
+                                       'api/kernels/',
+                                       kernel_id)
+            delete_req = requests.delete(delete_url)
+            if delete_req.status_code != 204:
+                QMessageBox.warning(
+                self,
+                _("Server error"),
+                _("The Jupyter Notebook server failed "
+                  "to shutdown the kernel. "
+                  "If you want to shutdown it you would "
+                  "have to do it manually."
+                  "<br><br>Status code {}").format(delete_req.status_code))

```

Comment:
Please remove this line, I don't think we need to show the response's status code.